ui/src/views/actions/thunks.ts

```diff
@@ -112,8 +115,50 @@ export const getViewForTimeMachine = (
         view,
       })
     )
+  } catch (error) {
+    console.error(error)
+    dispatch(notify(copy.getViewFailed(error.message)))
+    dispatch(setView(cellID, RemoteDataState.Error))
+  }
+}
+
+export const setQueryResultsByQueryID = (queryID: string) => (
+  dispatch,
+  getState: GetState
+): Promise<void> => {
+  try {
+    const state = getState()
+    const files = state.data.queryResultsByQueryID[hashCode(queryID)]
+    if (files) {
+      dispatch(setQueryResults(RemoteDataState.Done, files, null, null))
+      return
+    }
+    dispatch(executeQueries())
+  } catch (error) {
+    // if the files don't exist in the cache, we want to execute the query
+    console.error(error)
+    dispatch(executeQueries())
+  }
+}
+
+export const setQueryResultsForCell = (
+  dashboardID: string,
+  cellID: string,
+  timeMachineID: TimeMachineID
+) => (dispatch, getState: GetState): Promise<void> => {
+  try {
+    dispatch(getViewForTimeMachine(dashboardID, cellID, timeMachineID))
+    const state = getState()
+    const {view} = getActiveTimeMachine(state)
+    const queries = view.properties.queries.filter(({text}) => !!text.trim())
+    const queryID = get(queries, '[0].text', '')
+    if (queryID) {
```

Comment:
This should be where you lookup to see whether the queryID exists in your cache.

Reply:
in TimeSeries- we setQueryResults, here we should read it back from queryResults. 
so if `state.queryResults.Query.queryResultsByQueryID[queryID]` exists we should use it in the visualization, and not executeQueries. (you might need to look into how to use in the visualization)

Reply:
I think there might be a misunderstanding stemming from naming two different functions the same thing, since this dispatch should be dispatching the action above:

https://github.com/influxdata/influxdb/pull/18581/files#diff-d32f5a03b5c68c499b895f2222a64da9R125

and not the action to set the data

Reply:
yes, that is pretty confusing :) I don't think we need a separate `setQueryResultsByQueryID` function here, just roll what it does into this action.